aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2070,7 +2074,7 @@ Tensor numpy_T(const Tensor &self) {
 }
 
 Tensor view(const Tensor& self, IntArrayRef size) {
-  auto inferred_size = at::infer_size(size, self.numel());
+  at::DimVector inferred_size = at::infer_size_dv(size, self.numel());
```

Comment:
can you make `unflatten` and `reshape` use `infer_size_dv` and get rid of `infer_size`?

Reply:
Yes to the former (done!), sadly no to the latter - there are other callers. (The water is muddied by another `infer_size()` defined in `ExpandUtils.h/cpp`: some callers of our infer_size() wind up resolving incorrectly to that one if we remove ours, making it look like there are no callers to ours outside this file. But renaming our `DimVector`-returning `infer_size_dv()` to `infer_size()` provokes compile errors that reveal the callers - https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/static/ops.cpp#L29, https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/mkldnn/TensorShape.cpp#L47. Those are entangled in a way that will make them hard to port over.)